frame/instrumental-strategy-pablo/src/lib.rs

```diff
@@ -237,6 +247,29 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
+		/// Add vault_id to AssociatedVaults storage.
+		///
+		/// Emits [`AssociatedVault`](Event::AssociatedVault) event when successful.
+		#[pallet::weight(T::WeightInfo::associate_vault())]
+		pub fn associate_vault(
+			origin: OriginFor<T>,
+			vault_id: T::VaultId,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let access_right = Self::admin_accounts(who).ok_or(Error::<T>::NotAdminAccount)?;
+			ensure!(
+				access_right == AccessRights::Full ||
+					access_right == AccessRights::AssociateVaultId,
+				Error::<T>::NotEnoughAccessRights
+			);
+			// TODO(belousm): Ask Kevin do we need check that 'deposits_allowed'.
```

Comment:
I don't think we need this check. Since this action is permissioned it can be assumed that the caller didn't use a stopped vault and, if they did, it will become unstopped soon

Reply:
Got it!